lib/plek.rb

```diff
@@ -37,8 +35,12 @@ class NoConfigurationError < StandardError; end
   #   +GOVUK_APP_DOMAIN_EXTERNAL+ and if that is unavailable the parent domain
   #   will be used
   def initialize(domain_to_use = nil, external_domain = nil)
-    self.parent_domain = domain_to_use || env_var_or_dev_fallback("GOVUK_APP_DOMAIN", DEV_DOMAIN)
-    self.external_domain = external_domain || ENV["GOVUK_APP_DOMAIN_EXTERNAL"] || parent_domain
+    whitespace = " \t\r\n"
+    @parent_domain = domain_to_use || env_var_or_dev_fallback("GOVUK_APP_DOMAIN", DEV_DOMAIN)
+    @external_domain = external_domain || ENV["GOVUK_APP_DOMAIN_EXTERNAL"] || @parent_domain
+    @host_prefix = ENV["PLEK_HOSTNAME_PREFIX"] || ""
+    @unprefixable_hosts = ENV.fetch("PLEK_UNPREFIXABLE_HOSTS", "").delete(whitespace).split(",").to_set
+    @http_domains = ENV.fetch("PLEK_HTTP_DOMAINS", DEV_DOMAIN).delete(whitespace).split(",").to_set
```

Comment:
Also on this one, now I think I understand about the special `.` PLEK_HTTP_DOMAINS value. Do we expect this will be used in replatforming beyond that case? I'm just wondering if we're making this a bit complex if the value will be always a `.`.